examples/docs_snippets/docs_snippets/guides/dagster/dagster_pipes/pyspark/dagster_code.py

```diff
@@ -0,0 +1,92 @@
+# start_pipes_session_marker
+
+import os
+import subprocess
+from pathlib import Path
+from typing import Mapping, Sequence
+
+import boto3
+from dagster_aws.pipes import PipesS3ContextInjector, PipesS3MessageReader
+
+import dagster as dg
+
+LOCAL_SCRIPT_PATH = Path(__file__).parent / "script.py"
+
+
+@dg.asset
+def pipes_spark_asset(context: dg.AssetExecutionContext):
+    s3_client = boto3.client("s3")
+
+    bucket = os.environ["DAGSTER_PIPES_BUCKET"]
+
+    # upload the script to S3
+    # ideally, this should be done via CI/CD processes and not in the asset body
+    # but for the sake of this example we are doing it here
+    s3_script_path = f"{context.dagster_run.run_id}/pyspark_script.py"
+    s3_client.upload_file(LOCAL_SCRIPT_PATH, bucket, s3_script_path)
+
+    context_injector = PipesS3ContextInjector(
+        client=s3_client,
+        bucket=bucket,
+    )
+
+    message_reader = PipesS3MessageReader(
+        client=s3_client,
+        bucket=bucket,
+        # the following setting will configure the Spark job to collect logs from the driver
+        # and send them to Dagster via Pipes
+        include_stdio_in_messages=True,
+    )
+
```

Comment:
I would move `#end_pipes_session_marker` here -- I think it's a little confusing to split up the context manager between steps.